java/src/Ice/src/main/java/com/zeroc/IceInternal/ProxyOutgoingAsyncBaseI.java

```diff
@@ -303,37 +305,184 @@ protected boolean finished(boolean ok, boolean invoke) {
     return super.finished(ok, invoke);
   }
 
-  protected int handleException(com.zeroc.Ice.Exception exc) {
-    Holder<Integer> interval = new Holder<>();
-    _cnt = _proxy._handleException(exc, _handler, _mode, _sent, interval, _cnt);
-    return interval.value;
+  protected int handleException(com.zeroc.Ice.Exception ex) {
```

Comment:
Moved from `ObjectPrx`. Note that this differs from C++ and C#. In those, this code was moved to `RequestHandlerCache`, whereas I moved it to `OutgoingAsync` in Java. I think this is a better placement, since that is the _only_ place where this function is called. Moving it there lets these become fully private/inlined functions, instead of `protected`, and keeps the code close to where it's called from.

If this approved, I plan to update C++ and C# accordingly.